contract/src/main/kotlin/com/r3/corda/sdk/token/contracts/states/FungibleToken.kt

```diff
@@ -35,7 +35,13 @@ import net.corda.core.schemas.QueryableState
 open class FungibleToken<T : TokenType>(
         override val amount: Amount<IssuedTokenType<T>>,
         override val holder: AbstractParty
-) : FungibleState<IssuedTokenType<T>>, AbstractToken(), QueryableState {
+) : FungibleState<IssuedTokenType<T>>, AbstractToken<T>(), QueryableState {
+
+    override val tokenType: T get() = amount.token.tokenType
```

Comment:
Can remove if using definitions from `AbstractToken`.

Reply:
Ah the definitions for fungible token and non fungible token differ slightly.